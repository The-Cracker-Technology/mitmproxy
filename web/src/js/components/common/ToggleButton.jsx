import React  from 'react'
import PropTypes from 'prop-types'

ToggleButton.propTypes = {
    checked: PropTypes.bool.isRequired,
    onToggle: PropTypes.func.isRequired,
    text: PropTypes.string.isRequired
}

export default function ToggleButton({ checked, onToggle, text }) {
    return (
        <div className={"btn btn-toggle " + (checked ? "btn-primary" : "btn-default")} onClick={onToggle}>
            <i className={"fa fa-fw " + (checked ? "fa-check-square-o" : "fa-square-o")}/>
            &nbsp;
            {text}
        </div>
    )
}
