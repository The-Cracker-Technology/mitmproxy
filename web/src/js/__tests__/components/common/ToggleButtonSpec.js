import React from 'react'
import renderer from 'react-test-renderer'
import ToggleButton from '../../../components/common/ToggleButton'

describe('ToggleButton Component', () => {
    let mockFunc = jest.fn()

    it('should render correctly', () => {
        let checkedButton = renderer.create(
                <ToggleButton checked={true} onToggle={mockFunc} text="foo">
                    text
                </ToggleButton>),
        tree = checkedButton.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should handle click action', () => {
        let uncheckButton = renderer.create(
            <ToggleButton checked={false} onToggle={mockFunc} text="foo">
                text
            </ToggleButton>),
        tree = uncheckButton.toJSON()
        tree.props.onClick()
        expect(mockFunc).toBeCalled()
    })
})
