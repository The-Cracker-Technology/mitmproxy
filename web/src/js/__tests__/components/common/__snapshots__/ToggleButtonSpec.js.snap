// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton Component should render correctly 1`] = `
<div
  className="btn btn-toggle btn-primary"
  onClick={[Function]}
>
  <i
    className="fa fa-fw fa-check-square-o"
  />
  Â 
  foo
</div>
`;
